main.go

```diff
@@ -113,13 +113,15 @@ func main() {
 		}
 	}
 
+	defaultRestConfig := ctrl.GetConfigOrDie()
+	defaultKubeClient := kubernetes.NewForConfigOrDie(defaultRestConfig)
+
 	restConfig := ctrl.GetConfigOrDie()
 	restConfig.Timeout = time.Minute
-
-	kubeClient := kubernetes.NewForConfigOrDie(restConfig)
+	timeoutKubeClient := kubernetes.NewForConfigOrDie(restConfig)
```

Comment:
the two clients look a bit odd, but from reading @Mojachieee comments on slack I think I understand why. 😄

Reply:
Exactly `timeoutClient` used at one place, where there is no other option to give timeout :( Other places are working with timeout less client, but using time out context.